src/data.rs

```diff
@@ -196,6 +202,82 @@ impl Record {
         self
     }
 
+    /// Places a Value in the data based on the Expr accessor.
+    /// Only works with NestedColumn exprs.
+    pub fn put_expr(mut self, key: &Expr, value: Value) -> Result<Record, EvalError> {
+        match key {
+            Expr::NestedColumn { ref head, ref rest } => {
+                let mut root_record: &mut Value = if let Some(record) = self.data.get_mut(head) {
```

Comment:
hmm. I wonder if we could do something like:
1. construct a nestedcolumn that has the last item removed.
2. use the get machinery to load the container.
3. Use simpler machinery to insert (without multi level stuff required)

In general, I'm OK with this -- it's a little hacky but doing it this way doesn't make it any harder to refactor later, it's just an implementation detail.

Reply:
Hmm good idea. My original idea was to keep the logic together with the `get` function and make a similar `get_mut` gateway, but realized that `get_mut` didn't make a lot of sense for all the branches so put it elsewhere.

I think there's some potential refactoring we could do where instead of accepting an Expr for `input_column` / `output_column`, `Expr::NestedColumn` enclosed an inner type which was validated at the parser level and returned to the operator to work with `get` and `put_expr`.